### PR TITLE
adding entity courseId and parameter in mutation

### PR DESCRIPTION
**Pull request by Keyshal-211**
## Description of changes
Add course ID to content
Add queryByCourseID
Fix section position not being deterministic
Database must be adapted before merge

DO NOT MERGE UNTIL  GATEWAY READY
  
## How has this been tested?

Please describe the test strategy you followed. If not tested, please explain why.

- [x] automated unit test
- [x] automated integration test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
    
## Checklist before requesting a review

- [X] My code is easy to understand
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] My code fulfills all acceptance criteria
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation or
  the [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I made no breaking changes in the database schema or if so, I will perform a database migration

## Checklist for reviewer

- The code is easy to understand
- The code follows
  the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of
  this project
- The code is tested or if not, the reason is documented or discussed
- The added and existing tests reasonably cover the code change
- The code has no breaking changes in the database schema or if so, the assignee is aware of it and
  will perform a database migration

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/content_service/persistence/entity/StageEntity.java [hunk 1]
```diff
@@ -8,6 +8,9 @@
 import java.util.UUID;
 
 @Entity(name = "Stage")
+@Table(indexes = {
+        @Index(name = "idx_stage_section_id", columnList = "section_id")
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/content_service/persistence/entity/StageEntity.java:12]: foreign keys do not automatically lead to an index

**Review by myluki2000 — CHANGES_REQUESTED** [review]
Looks good at a glance. But this can only be merged when everything is ready in the other services & gateway.

When everything is ready I will do some very small changes (graphql schema) and then merge.

**Comment by PaulBredl** [comment]
> Looks good at a glance. But this can only be merged when everything is ready in the other services & gateway.
> 
> When everything is ready I will do some very small changes (graphql schema) and then merge.

warn me though because of the DB migration

File: src/main/java/de/unistuttgart/iste/gits/content_service/controller/ContentController.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public MediaContent updateMediaContent(@Argument UpdateMediaContentInput input,
     }
 
     @MutationMapping
-    public Assessment createAssessment(@Argument CreateAssessmentInput input, @Argument UUID courseId) {
+    public Assessment _internal_createAssessment(@Argument CreateAssessmentInput input, @Argument UUID courseId) {
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/content_service/controller/ContentController.java:73]: This is not the Java naming convention. Please keep the old name and specify the correct name in the MutationMapping annotation

**Review by PaulBredl — CHANGES_REQUESTED** [review]
@myluki2000

File: src/main/java/de/unistuttgart/iste/gits/content_service/controller/ContentController.java (hunk 2 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/content_service/controller/ContentController.java:73]: @Paul I did it

**Review by PaulBredl — APPROVED** [review]

**Review by myluki2000 — APPROVED** [review]

**Pull request merged**